### PR TITLE
Fix java17 PostCommit ARM test config issue

### PR DESCRIPTION
**Pull request by celeste-zeng**
Java17 test suite failed because of test configuration issue:
image

which is caused by the --release flag added here: https://github.com/apache/beam/blob/a434a31507895ecfef3be94bdc1729cf376ebf56/buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy#L1489
This PR:
- Replaced the --release flag with -source -target to compile, reference: https://github.com/eclipse-tycho/tycho/issues/619#issuecomment-1285022312
- Removed Werror flag that causes all compiler warnings to be treated as errors, otherwise the tests would fail to compile.
Now All versions of PostCommit Java Dataflow Example ARM tests passed.
------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

**Comment by celeste-zeng** [comment]
Run Java_Examples_Dataflow_ARM PostCommit 17

**Comment by celeste-zeng** [comment]
Run Java_Examples_Dataflow_ARM PostCommit 8
Run Java_Examples_Dataflow_ARM PostCommit 11
Run Java_Examples_Dataflow_ARM PostCommit 17

**Comment by celeste-zeng** [comment]
Run Java_Examples_Dataflow_ARM PostCommit 17

**Comment by celeste-zeng** [comment]
Run Java_Examples_Dataflow_ARM PostCommit 17

**Comment by celeste-zeng** [comment]
Run Java_Examples_Dataflow_ARM PostCommit 11

File: buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy [hunk 1]
```diff
@@ -1073,7 +1074,6 @@ class BeamModulePlugin implements Plugin<Project> {
         options.compilerArgs += ([
           '-parameters',
           '-Xlint:all',
-          '-Werror'
```
Comment (tvalentyn) [inline buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy:1076]: What were the warnings?

cc: @kennknowles

Comment (celeste-zeng) [reply buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy:1076]: image

Comment (Abacn) [reply buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy:1076]: please just change the naming of this interface. It's in Snippets (code example) no concern for breaking change

**Comment by celeste-zeng** [comment]
Run Java_Examples_Dataflow_ARM PostCommit 17

**Comment by celeste-zeng** [comment]
Run Java_Examples_Dataflow_ARM PostCommit 17

**Comment by celeste-zeng** [comment]
Run Spotless PreCommit

**Comment by celeste-zeng** [comment]
Run Java_Examples_Dataflow_ARM PostCommit 17

**Comment by celeste-zeng** [comment]
Run Java PreCommit

**Comment by celeste-zeng** [comment]
Run Portable_Python PreCommit

**Comment by celeste-zeng** [comment]
Run Python_Coverage PreCommit

**Comment by celeste-zeng** [comment]
Run Java_Examples_Dataflow_Java11_PreCommit

**Comment by celeste-zeng** [comment]
Run Java PreCommit

**Comment by celeste-zeng** [comment]
Run Java PreCommit

File: buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy (hunk 1 above)
Comment (celeste-zeng) [reply buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy]: Done, tested and passed.

**Review by Abacn — APPROVED** [review]

**Comment by Abacn** [comment]
It appears the github action failure is not flake: https://github.com/apache/beam/actions/workflows/beam_PreCommit_Java_Examples_Dataflow_Java11.yml while the same test on jenkins passed
https://ci-beam.apache.org/view/PreCommit/job/beam_PreCommit_Java_Examples_Dataflow_Java11_Cron/

**Pull request merged**